common/components/ui/DropdownShell.tsx

```diff
@@ -14,7 +14,7 @@ interface State {
   expanded: boolean;
 }
 
-export default class DropdownComponent extends PureComponent<Props, State> {
+export default class DropdownComponent extends Component<Props, State> {
```

Comment:
Since DropdownComponent takes in a function to render its content, it's not pure. The reference to the function may not change, but its return value could, which it will fail to re-render with since it won't rerun the render functions.